db/storage.go

```diff
@@ -0,0 +1,148 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/big"
+	"github.com/google/uuid"
+)
+
+type StorageLog struct {
+	DealUUID  uuid.UUID
+	CreatedAt time.Time
+	PieceSize abi.PaddedPieceSize
+	Text      string
+}
+
+type StorageDB struct {
+	db *sql.DB
+}
+
+func NewStorageDB(db *sql.DB) *StorageDB {
+	return &StorageDB{db: db}
+}
+
+func (s *StorageDB) Tag(ctx context.Context, dealUuid uuid.UUID, pieceSize abi.PaddedPieceSize) error {
+	ps := big.NewIntUnsigned(uint64(pieceSize))
+
+	qry := "INSERT INTO StorageTagged (DealUUID, CreatedAt, PieceSize) "
+	qry += "VALUES (?, ?, ?)"
+	values := []interface{}{dealUuid, time.Now(), ps.String()}
+	_, err := s.db.ExecContext(ctx, qry, values...)
+	return err
+}
+
+func (s *StorageDB) Untag(ctx context.Context, dealUuid uuid.UUID) (abi.PaddedPieceSize, error) {
+	qry := "SELECT PieceSize FROM StorageTagged WHERE DealUUID = ?"
+	row := s.db.QueryRowContext(ctx, qry, dealUuid)
+
+	ps := &bigIntFieldDef{f: new(big.Int)}
```

Comment:
I wonder if we can find a simpler way to convert from string to uint64 🤔

Reply:
We can use `strconv.FormatUint64(val, 10)` or `big.Int`, I don't think it makes much difference.